server/lib/api/users.js

```diff
@@ -97,3 +106,33 @@ async function resetPassword(ctx, next) {
     ctx.status = 204
   })
 }
+
+async function verifyEmail(ctx, next) {
```

Comment:
They are all arbitrary, yes. Generally you can find/choose similar values for rate-limiting similar things. Things that cause emails to be sent should be rate-limited most stringently, things that are mainly to prevent brute-forcing (and have consequences for other users) can allow far more requests. It's a judgment call: how much do you expect a particular entity to call something in the normal case? How many calls would be considered very exceptional? Would we expect the exceptional case to continue for a prolonged period, or only in a single burst (and then return to the normal rate)?